### PR TITLE
Auto-generate Actions.md based on source

### PR DESCRIPTION
**Pull request by KrauseFx**
Opening the actions screen of [docs.fastlane.tools](https://docs.fastlane.tools) will now unfold the available categories:

screenshot 2016-08-29 19 23 42

---

The user can easily click on the sidebar to select a category

screenshot 2016-08-29 19 23 48

You can preview the most up to date auto-generated actions doc overe [here](https://github.com/fastlane/fastlane/blob/auto-docs-generate/fastlane/docs/ActionsAuto.md). This only includes the migrated actions.

To "migrate" an action to be included in the docs, you have to do the following
- Check out this branch
- Run `bundle exec rspec -t now` to get a list of actions that were not migrated yet
- Add a category to the action, you can find the available categories in [`action.rb`](https://github.com/fastlane/fastlane/blob/auto-docs-generate/fastlane/lib/fastlane/action.rb#L5).

``` ruby
      def self.category
        :misc
      end
```
- Open [`Actions.md`](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Actions.md) and move over all relevant documentation that's not in the action's class yet over. As soon as it's more than 80 characters, have the docs in `self.details`. You can have multi-line statements, just do it like this:

``` ruby

def self.details
  [
   "This will automatically tag your build with the following format: `<grouping>/<lane>/<prefix><build_number>`, where:",
   "- `grouping` is just to keep your tags organised under one 'folder', defaults to 'builds'",
   "- `lane` is the name of the current fastlane lane",
   "- `prefix` is anything you want to stick in front of the version number, e.g. 'v'",
   "- `build_number` is the build number, which defaults to the value emitted by the `increment_build_number` action",
   "",
   "For example for build 1234 in the 'appstore' lane it will tag the commit with `builds/appstore/1234`"
  ].join("\n")
end
```
- Add sample code calls, at least one call should be included for every action. Have one example that's the easiest call, and then one or more advanced ones:

``` ruby
def self.example_code
   [
     'add_git_tag # simple tag with default values',
     'add_git_tag(
       grouping: "fastlane-builds",
       prefix: "v",
       build_number: 123
     )',
     '# Alternatively, you can specify your own tag.
     add_git_tag(
       tag: "my_custom_tag"
     )'
   ]
 end
```

An example change that migrates some of the actions https://github.com/fastlane/fastlane/pull/5743/commits/a70a9b53d98c2265bebce960372dc29c0e62f5f4

To generate the markdown file, just run

```
bundle exec rake generate_docs
```

Please add a comment before updating actions so that everybody knows have a read/write lock on them

The tests of this PR will fail until we migrated all the actions ♻️

### CODE REVIEW

**Comment by ohwutup** [comment]
This looks really sweet. Should we add tests for this at all?

File: fastlane/lib/fastlane/actions/zip.rb [hunk 1]
```diff
@@ -40,6 +40,16 @@ def self.available_options
         ]
       end
 
+      def self.example_code
+        [
+          "zip",
+          "zip(
```
Comment (ohwutup) [inline fastlane/lib/fastlane/actions/zip.rb]: Based on the code in the `erb` file, I am a bit confused why we would want to have two entries in this example. It looks like the `erb` file is iterating over the things in this list and putting them in that code block, is there a reason why there is `'zip'` and `'zip(...` in there? 

I could be missing something obvious here 😄

Comment (mfurtak) [inline fastlane/lib/fastlane/actions/zip.rb]: Just guessing, but perhaps they're two different code usage examples? (one plain, and one with params)

File: fastlane/lib/fastlane/documentation/markdown_docs_generator.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    def work
+      self.categories = {}
+      ActionsList.all_actions do |action|
+        self.categories['Misc'] ||= []
+        self.categories['Misc'] << action
+      end
+    end
+
+    def save_to_ssd!(target_path: "docs/ActionsAuto.md")
```
Comment (mfurtak) [inline fastlane/lib/fastlane/documentation/markdown_docs_generator.rb]: This method name is a little strang :smile: `generate!` would probably be easier to understand and guess 👍

Comment (KrauseFx) [inline fastlane/lib/fastlane/documentation/markdown_docs_generator.rb]: Haha, yah, all the methods are usually called `save_to_disk!` so I figured `ssd` might be more appropriate 😜

File: fastlane/lib/fastlane/actions/zip.rb (hunk 1 above)
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/zip.rb]: Yeah, different variants, easy example, more advanced one and special case. Whatever is necessary to explain the actoin

**Comment by KrauseFx** [comment]
Picking this up now 👍

**Comment by KrauseFx** [comment]
Ok, migrating some more actions now

**Comment by KrauseFx** [comment]
Done for now, I'm finished from `a` to `create_pull_request` 👍

File: fastlane/lib/fastlane/actions/commit_version_bump.rb [hunk 3]
```diff
@@ -155,13 +155,43 @@ def self.available_options
...
+          'commit_version_bump',
+          'commit_version_bump(
+            message: "Version Bump",                    # create a commit with a custom message
+            xcodeproj: "./path/to/MyProject.xcodeproj", # optional, if you have multiple Xcode project files, you must specify your main project here
+          )'
+        ]
+      end
+
+      def self.category
+        :version_control
```
Comment (asfalcone) [inline fastlane/lib/fastlane/actions/commit_version_bump.rb]: should this be `:version_control` or `:source_control`?

Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/commit_version_bump.rb]: Nice catch 👍

File: fastlane/lib/fastlane/actions/ensure_xcode_version.rb [hunk 4]
```diff
@@ -58,6 +58,14 @@ def self.authors
         ["JaviSoto"]
```
Comment (KrauseFx) [inline fastlane/lib/fastlane/actions/ensure_xcode_version.rb]: A wild @JaviSoto appeared

Comment (JaviSoto) [inline fastlane/lib/fastlane/actions/ensure_xcode_version.rb]: haha best fastlane action

**Comment by KrauseFx** [comment]
Working on this again 👍

File: fastlane/lib/fastlane/documentation/markdown_docs_generator.rb [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      require 'faraday'
+      require 'json'
+
+      unless @launches
+        conn = Faraday.new(ENHANCER_URL)
+        conn.basic_auth(ENV["ENHANCER_USER"], ENV["ENHANCER_PASSWORD"])
+        @launches = JSON.parse(conn.get('/index.json').body)
+      end
+
+      found = @launches.find { |c| c["action"] == action_name.to_s }
```
Comment (mfurtak) [inline fastlane/lib/fastlane/documentation/markdown_docs_generator.rb]: Hm, I guess whatever I did with Rubocop still didn't fix this 😫

**Comment by KrauseFx** [comment]
Resolved merge conflicts 👍

**Review by ohayon — APPROVED** [review]
💯

**Pull request merged**